kore/src/Kore/Builtin/AssociativeCommutative.hs

```diff
@@ -420,11 +421,14 @@ Return 'Nothing' if there are any duplicate keys.
 
  -}
 updateConcreteElements
-    :: Ord key
-    => Map key value
-    -> [(key, value)]
-    -> Maybe (Map key value)
-updateConcreteElements = Foldable.foldrM (uncurry insertMissing)
+    :: Map (TermLike Concrete) value
+    -> [(TermLike Concrete, value)]
+    -> Maybe (Map (TermLike Concrete) value)
```

Comment:
@virgil-serbanuta I added the check in this function as well (I hope that makes sense). Judging by the name of this function and its context it doesn't seem like we ever intend it to be used outside of the case where we have a `TermLike Concrete` as the key type. Do you think this seems reasonable or do we actually need a more general type here?

Reply:
I'm not sure which kind of generalization you mean. To pick one of them, this function was meant only for concrete elements. We also have `updateAbstractElements` for updating the others. Checking this here makes sense.

Reply:
Alright. I was just asking since I changed `key` to `TermLike Concrete` to make this work. Thanks!

Reply:
Oh, I see! Well, I'm not going to complain if it compiles...